pkg/store/matchers.go

```diff
@@ -33,3 +35,64 @@ func translateMatchers(ms []storepb.LabelMatcher) (res []*labels.Matcher, err er
 	}
 	return res, nil
 }
+
+// matchersToMetric converts label matchers to metric format.
+func matchersToMetric(ms []storepb.LabelMatcher) string {
+	var res string
+	var nameMatchers []storepb.LabelMatcher
+	var otherMatchers []string
+	for _, m := range ms {
+		if m.Name == "__name__" {
+			nameMatchers = append(nameMatchers, m)
+		} else {
+			otherMatchers = append(otherMatchers, labelMatcherToString(m))
+		}
+	}
+
+	res = strings.Join(otherMatchers, ", ")
+
+	if len(nameMatchers) == 1 {
+		// If there is only one equal name matcher, we can put it outside the {}.
+		if nameMatchers[0].Type == storepb.LabelMatcher_EQ {
+			if len(otherMatchers) == 0 {
+				return nameMatchers[0].Value
+			}
+			return nameMatchers[0].Value + "{" + res + "}"
+		} else {
+			if len(otherMatchers) == 0 {
+				return "{" + labelMatcherToString(nameMatchers[0]) + "}"
+			} else {
+				return "{" + labelMatcherToString(nameMatchers[0]) + ", " + res + "}"
+			}
+		}
+	}
+
+	// More than one name matcher, we need to put them all inside {}.
+	for i, v := range nameMatchers {
+		n := labelMatcherToString(v)
+		if i != len(nameMatchers)-1 {
+			n += ", "
+		}
+		res += n
+	}
+
+	return "{" + res + "}"
+}
+
+func labelMatcherToString(m storepb.LabelMatcher) string {
```

Comment:
Maybe let's reuse https://github.com/thanos-io/thanos/blob/master/pkg/store/matchers.go#L9 here + https://godoc.org/github.com/prometheus/tsdb/labels#Matcher ?

Reply:
wow... I didn't even find this is already implemented 😢 Thanks, I will switch to it.